kas-fleet-manager/deploy-kas-fleet-manager.sh

```diff
@@ -141,35 +92,60 @@ deploy_kasfleetmanager() {
   create_kas_fleet_manager_namespace
 
   echo "Deploying KAS Fleet Manager Database..."
-  ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/db-template.yml | oc apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
+  ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/db-template.yml | ${OC} apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
   echo "Waiting until KAS Fleet Manager Database is ready..."
   time timeout --foreground 3m bash -c "until ${OC} get pods -n ${KAS_FLEET_MANAGER_NAMESPACE}| grep kas-fleet-manager-db | grep -v deploy | grep -q Running; do echo 'database is not ready yet'; sleep 10; done"
 
+  create_kasfleetmanager_service_account
+  create_kasfleetmanager_pull_credentials
+
   echo "Deploying KAS Fleet Manager K8s Secrets..."
   ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/secrets-template.yml \
-    -p OCM_SERVICE_CLIENT_ID="dummyclient" \
-    -p OCM_SERVICE_CLIENT_SECRET="dummysecret" \
+    -p OCM_SERVICE_CLIENT_ID="" \
+    -p OCM_SERVICE_CLIENT_SECRET="" \
+    -p OCM_SERVICE_TOKEN="${OCM_SERVICE_TOKEN}" \
     -p OBSERVABILITY_CONFIG_ACCESS_TOKEN="${OBSERVABILITY_CONFIG_ACCESS_TOKEN}" \
     -p MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
     -p MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
     -p MAS_SSO_CRT="${MAS_SSO_CRT}" \
     -p KAFKA_TLS_CERT="${KAFKA_TLS_CERT}" \
     -p KAFKA_TLS_KEY="${KAFKA_TLS_KEY}" \
-    -p DATABASE_HOST="$(${KUBECTL} get service/kas-fleet-manager-db -o jsonpath="{.spec.clusterIP}")" \
+    -p KUBE_CONFIG="$(${OC} config view --minify --raw | base64 -w0)" \
+    -p IMAGE_PULL_DOCKER_CONFIG=$(${OC} get secret ${KAS_FLEET_MANAGER_IMAGE_PULL_SECRET_NAME} -n ${KAS_FLEET_MANAGER_NAMESPACE} -o jsonpath="{.data.\.dockerconfigjson}") \
     | ${OC} apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
   echo "Deploying KAS Fleet Manager Envoy ConfigMap..."
   ${OC} apply -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/envoy-config-configmap.yml -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
-  create_kasfleetmanager_service_account
-  create_kasfleetmanager_pull_credentials
-
   echo "Deploying KAS Fleet Manager..."
   OCM_ENV="development"
 
+  SERVICE_PARAMS=${DIR_NAME}/kas-fleet-manager-params.env
+  > ${SERVICE_PARAMS}
+
+  if [ -n "${OCM_SERVICE_TOKEN}" ] ; then
+      PROVIDER_TYPE="ocm"
+      ENABLE_OCM_MOCK="false"
+      CLUSTER_STATUS="cluster_provisioned"
+      echo 'AMS_URL="https://api.stage.openshift.com"' >> ${SERVICE_PARAMS}
+      echo 'OCM_URL="https://api.stage.openshift.com"' >> ${SERVICE_PARAMS}
+      echo 'ENABLE_KAFKA_SRE_IDENTITY_PROVIDER_CONFIGURATION="false"' >> ${SERVICE_PARAMS}
```

Comment:
I think this also has to be applied to `standalone` mode. 
If not disabled there as well, we'll try to configure the IDP for the standalone mode cluster as well.

Reply:
Interesting, I didn't notice it in `standalone` mode, but I now see that with `standalone` I can log in with _either_ Kafka_SRE or kubeadmin whereas with OCM the kubeadmin option was missing.

Reply:
Yes; The configuration is there for both modes. With this flag, when it is turned off, the configuration is disabled for all kinds of clusters (ocm and non ocm). I think we should just turn it off for standalone too here - as that means one less reconciliation step in KFM.
 
> whereas with OCM the kubeadmin option was missing.

I think ("to be confirmed") the OCM api does something to hide kubeadmin option when the IDP is configured.

Reply:
Updated, the value of `ENABLE_KAFKA_SRE_IDENTITY_PROVIDER_CONFIGURATION` will now always be set to false.